cabal.haskell-ci

```diff
@@ -1,14 +1,7 @@
 no-tests-no-benchmarks: False
 unconstrained:          False
+hlint:                  True
+hlint-yaml:             .hlint.yaml
```

Comment:
Hm, I've never needed to specify `.hlint.yaml` in `cabal.haskell-ci` before, as `hlint` automatically detects its presence. What is the advantage of specifying it manually?

Reply:
it doesn't detect, as the build is not done from repository root. I should change `haskell-ci` to detect the default `.hlint.yaml` automatically, but for now this is needed to fix the CI-run of HLint.

Reply:
Sorry, I wasn't very clear. In other repos where `hlint` is used, `.hlint.yaml` is included in the `extra-source-files`, so it will be included in the sdist. If this is done, then one shouldn't need to specify `.hlint.yaml` again in `cabal.haskell-ci`.

I suppose this comes down to whether you want to distribute your `hlint` configuration with sdists or not. I don't have a strong preference one way or the other, although it would be good to pick one convention and use it consistently.

Reply:
I don't think distributing hlint config adds much, neither it costs much either.

I'll fix `haskell-ci` issue, so you can make a choice, I'm fine with either way.

Reply:
I'm not entirely sure how this `hlint-yaml` setting would work for projects where there are multiple `.hlint.yaml` files, such as `lens`. (See https://github.com/ekmett/lens/pull/960#issuecomment-753312466.)

Reply:
I'd use just single `hlint.yaml` for all packages in project. I don't remember if something bad happens with `includes` (`cpphs` just skips missing includes, so there is a warning), *yet* `haskell-ci` can be taught to pass `include-dirs` from package definition.

But this is my POV, I use the same `.hlint.yaml` for all my own packages, which I occasionally update :)

Reply:
OK. If I'm reading https://github.com/ekmett/linear/pull/158#discussion_r554584204, then you're planning to make some change to `haskell-ci` which affects this? If so, I'll wait for an indication of some sort before proceeding here.

Reply:
I will, but not immediately. So I'd suggest we merge this now.

Reply:
I will add `.hlint,yaml` to extra-source-files.